lib/http1.c

```diff
@@ -586,9 +648,18 @@ static void cleanup_connection(struct st_h2o_http1_conn_t *conn)
         return;
     }
 
+    if (conn->req.proceed_req) {
```

Comment:
I'd appreciate it if you could add `!= NULL`. We apply operators so that the result would be a boolean, unless the input (being an integral value) is considered a boolean (due to lack of a built-in boolean type in C).

Reply:
Adressed in c3bcd0d thank you.